libs/ml/tests/ml/optimisation/optimisers.cpp

```diff
@@ -158,25 +164,26 @@ TYPED_TEST(OptimisersTest, sgd_optimiser_training_2D)
 
   // Prepare model
   std::string                                  input_name;
+  std::string                                  label_name;
   std::string                                  output_name;
   std::shared_ptr<fetch::ml::Graph<TypeParam>> g =
-      PrepareTestGraph<TypeParam>(4, 2, input_name, output_name);
+      PrepareTestGraph<TypeParam>(4, 2, input_name, label_name, output_name);
 
   // Prepare data and labels
   TypeParam data;
   TypeParam gt;
   PrepareTestDataAndLabels2D(data, gt);
 
   // Initialize Optimiser
-  fetch::ml::optimisers::SGDOptimiser<TypeParam, fetch::ml::ops::MeanSquareError<TypeParam>>
-      optimiser(g, {input_name}, output_name, learning_rate);
+  fetch::ml::optimisers::SGDOptimiser<TypeParam> optimiser(g, {input_name}, label_name, output_name,
+                                                           learning_rate);
 
   // Do 2 optimiser steps
   optimiser.Run({data}, gt);
   DataType loss = optimiser.Run({data}, gt);
 
   // Test loss
-  EXPECT_NEAR(static_cast<double>(loss), 39.3290558, 1e-4);
+  EXPECT_NEAR(static_cast<double>(loss), 78.658111, 1e-4);
```

Comment:
these should not change. the division by 2 in the loss function should still take place in forward, not backward